dali/operators/reader/loader/filesystem.cc

```diff
@@ -22,14 +23,17 @@
 #include <string>
 #include <utility>
 #include <vector>
-
+#include "dali/core/call_at_exit.h"
 #include "dali/core/error_handling.h"
-#include "dali/operators/reader/loader/filesystem.h"
 #include "dali/operators/reader/loader/utils.h"
 
 namespace dali {
 namespace filesystem {
 
+inline bool starts_with(const std::string &str, const char *prefix) {
+  return str.rfind(prefix, 0) == 0;
```

Comment:
This is quite tricky - please add a comment explaining how it works.

Reply:
done